libcxx/include/__fwd/string.h

```diff
@@ -45,6 +47,10 @@ class _LIBCPP_TEMPLATE_VIS allocator;
 template <class _CharT, class _Traits = char_traits<_CharT>, class _Allocator = allocator<_CharT> >
 class _LIBCPP_TEMPLATE_VIS basic_string;
 
+// TODO: This could be extended to also allow custom allocators and possibly custom chars and char traits
+template <class _CharT>
+struct __libcpp_is_trivially_relocatable<basic_string<_CharT>> : is_fundamental<_CharT> {};
```

Comment:
We shouldn't allow all fundamental types as a `basic_string` specialization. Please limit it to the character types.

Reply:
True, but this seems to imply we allow other types and it might give the wrong impression to readers of this code.

Reply:
Here we could instead say:

```
template <class _CharT>
struct __libcpp_is_trivially_relocatable<basic_string<_CharT>> : __is_primary_template<basic_string<_CharT>> {};
```

Since `_CharT` must be trivial, `_CharT` has to be trivially relocatable. The only thing that matters is that we don't take for granted that user-defined specializations of `std::basic_string` are trivially relocatable.